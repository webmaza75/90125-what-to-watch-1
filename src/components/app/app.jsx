import React, {Fragment} from 'react';
import {Switch, Route} from 'react-router-dom';

import Main from '../main/main.jsx';
import SignIn from '../sign-in/sign-in.jsx';
import withPrivateRouter from '../../hocs/with-private-router/with-private-router.js';
import withPublicRouter from '../../hocs/with-public-router/with-public-router.js';
import MyList from '../my-list/my-list.jsx';
import MovieDetails from '../movie-details/movie-details.jsx';
import GlobalIcons from '../global-icons/global-icons.jsx';

const PrivateRouterWrapper = withPrivateRouter(MyList);
const PublicRouterWrapper = withPublicRouter(SignIn);


const App = () => (
  <Switch>
    <Fragment>
      <GlobalIcons />
      <Route path={`/`} exact component={Main} />
      <Route path={`/login`} component={PublicRouterWrapper} />
      <Route path={`/mylist`} component={PrivateRouterWrapper} />
      <Route path={`/film/:id`} exact component={MovieDetails} />
    </Fragment>
  </Switch>
);

export default App;
