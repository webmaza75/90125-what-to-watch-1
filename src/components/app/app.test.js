import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {App} from './app.jsx';

it(`App correctly renders`, () => {
  const renderer = new ShallowRenderer();
  const result = renderer
    .render(<App
      onCheckUser={jest.fn()}
      isLoadedUserInfo={true}
    />);

  expect(result).toMatchSnapshot();
});
