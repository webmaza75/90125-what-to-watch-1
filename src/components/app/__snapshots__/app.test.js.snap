// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders 1`] = `
<Switch>
  <React.Fragment>
    <GlobalIcons />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Main)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPublicRouter)",
          "type": [Function],
        }
      }
      path="/login"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPrivateRouter)",
          "type": [Function],
        }
      }
      path="/mylist"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPrivateRouter)",
          "type": [Function],
        }
      }
      path="/film/:id/review"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MovieDetails)",
          "type": [Function],
        }
      }
      exact={true}
      path="/film/:id"
    />
  </React.Fragment>
</Switch>
`;
